backend/src/common/utils/helper.ts

```diff
@@ -0,0 +1,73 @@
+import { open } from 'node:fs/promises';
+import { ANSWER_APPROXIMATELY, ANSWER_CORRETLY, ANSWER_WRONG, MINIMUM_CHAR_WRONG } from 'src/modules/socket/constant';
+const fs = require('fs');
+
+const PATH_FILE_LANGUAGE = 'src/common/files/language.txt';
+const PATH_FOLDER_AVATAR = 'src/common/public/avatars';
+
+export const randomString = (length: number): string => {
+  const result: string = (Math.random() + 1).toString(36).substring(length);
+  return result;
+}
+
+export const extractIdRoom = (codeRoom: string): number => {
+  const idRoom = codeRoom.split("_")[codeRoom.split("_").length - 1];
+  return Number.parseInt(idRoom);
+}
+
+
+type Language = {
+  code: string;
+  name: string;
+};
+
+export const getLanguages = async (): Promise<Array<Language>> => {
+  const languages: Array<Language> = [];
+  const languageFile = await open(PATH_FILE_LANGUAGE);
+
+  for await (const line of languageFile.readLines()) {
+    const [code, name] = line.split(',');
+    languages.push({
+      code: code,
+      name: name,
+    });
+  }
+
+  await languageFile.close();
+
+  return languages;
+}
+
+export const checkTypeAnswer = (answerRound: string, answerUser: string): number => {
+  if (answerRound === answerUser) {
+    return ANSWER_CORRETLY;
+  }
+
+  if (answerRound.length !== answerUser.length && answerUser.includes(answerRound)) {
+    return ANSWER_APPROXIMATELY;
+  }
+
+  let countCharCorretly = 0;
+  for (let index: number = 0; index < answerRound.length; index++) {
+    if (answerRound[index] === answerUser[index]) {
+      countCharCorretly++;
+    }
+  }
```

Comment:
Vậy cho đẹp trai :)))
```typescript
  const countCharCorrectly = [...answerRound].filter((char, index) => char === answerUser[index]).length;
```

Code JS thì dùng `filter`, `map`, `reduce`, ... cho đẹp, tránh dùng `for...`